doc/tutorials/introduction/building_fastcv/building_fastcv.markdown

```diff
@@ -0,0 +1,248 @@
+
+About FastCV
+------------
+
+FastCV provides two main features to computer vision application developers:
+- First, it provides a library of frequently used computer vision (CV) functions, optimized to run efficiently on a wide variety of Qualcomm’s Snapdragon devices.
+- Second, it provides a clean processor-agnostic hardware acceleration API, under which chipset vendors can hardware accelerate FastCV functions on Qualcomm’s Snapdragon hardware.
+
+FastCV is released as a unified binary, a single binary containing two implementations of the algorithms:
+- The first implementation runs on Arm® architecture and is referred to as FastCV for Arm architecture.
+- The second implementation runs only on Qualcomm® Snapdragon™ chipsets and is called FastCV for Snapdragon.
+FastCV library is Qualcomm proprietary and provides faster implementation of CV algorithms on various hardware as compared to other CV libraries.
+
+OpenCV Acceleration with FastCV HAL and Extensions
+--------------------------------------------------
+
+1. Accelerates OpenCV APIs using FastCV as the backend, resulting in improved performance.
+2. Enhances OpenCV functionality with FastCV-based HAL and Extension APIs, which were not available in earlier versions that only included the default OpenCV library.
+3. CV applications can be developed using the standard OpenCV APIs. The OpenCV library invokes FastCV HAL APIs, which in turn call FastCV algorithms.
+
+
+![](fastcv_hal_extns.png)
+
+Supported Platforms
+-------------------
+
+1. Android : Qualcomm Chipsets with the Android from Snapdragon 8 Gen 1 onwards(https://www.qualcomm.com/products/mobile/snapdragon/smartphones#product-list)
+2. Linux   : Qualcomm Linux Program related boards mentioned in [Hardware](https://www.qualcomm.com/developer/software/qualcomm-linux/hardware)
+
+Compiling OpenCV with FastCV for Android
+----------------------------------------
+
+1. **Follow Wiki page for OpenCV Compilation** : https://github.com/opencv/opencv/wiki/Custom-OpenCV-Android-SDK-and-AAR-package-build
+
+ Once the OpenCV repository code is cloned into the workspace, Please add `-DWITH_FASTCV=ON` flag to cmake vars as below to arm64 entry in **opencv\platforms\android\ndk-18-api-level-24.config.py**  to enable FastCV HAL/Extenstions compilation.
+
+ ```
+  ABI("3", "arm64-v8a", None, 24, cmake_vars=dict(WITH_FASTCV='ON')),
+ ```
+2. Remaining steps can be followed as mentioned in [the wiki page](https://github.com/opencv/opencv/wiki/Custom-OpenCV-Android-SDK-and-AAR-package-build)
+
+Compiling OpenCV with FastCV for Qualcomm Linux
+-----------------------------------------------
+
+1. Install eSDK by following [Qualcomm® Linux Documentation](https://docs.qualcomm.com/bundle/publicresource/topics/80-70017-51/install-sdk.html?vproduct=1601111740013072&version=1.3&facet=Qualcomm%20Intelligent%20Multimedia%20Product%20(QIMP)%20SDK&state=preview)
```

Comment:
Looks like the link is not correct. It does not work for me. Proper location: https://docs.qualcomm.com/bundle/publicresource/topics/80-70017-51/install-sdk.html